mm2src/coins/z_coin/storage/walletdb/mod.rs

```diff
@@ -33,27 +46,115 @@ pub struct WalletDbShared {
     ticker: String,
 }
 
+/// Calculates the starting block height based on a given date and the current block height.
+///
+/// # Arguments
+/// * `date`: The date in seconds representing the desired starting date.
+/// * `current_block_height`: The current block height at the time of calculation.
+///
+#[cfg(not(target_arch = "wasm32"))]
+fn calculate_starting_height_from_date(date: u32, current_block_height: u64) -> Result<u32, String> {
+    let blocks_prod_per_day = 24 * 60;
+    let current_time_s = now_sec();
+
+    let date = date as u64;
```

Comment:
I think it will be easy to convert from ISO format to unix from the GUI side so I don't think we need to make this complex to handle. Just `height` and `date` in unix format should be enough

Reply:
> I think it will be easy to convert from ISO format to unix from the GUI side so I don't think we need to make this complex to handle.

I agree, but let's wait for GUI devs opinion on this.